site/hooks/useRegisterToken.js

```diff
@@ -0,0 +1,54 @@
+import { useCallback } from 'react'
+import { useWeb3React } from '@web3-react/core'
+import defaultTokenList from '@uniswap/default-token-list'
+import { injected } from '../utils/connectors'
+
+export const useRegisterToken = function ({ symbol }) {
+  const { account, chainId, connector } = useWeb3React()
+
+  const registerToken = useCallback(
+    function () {
+      const storageKey = `isTokenRegistered-${symbol}-${account}-${chainId}`
+      const { ethereum, localStorage } = window
+      const { symbol: tokenSymbol, address, decimals, logoURI } =
+        defaultTokenList.tokens.find(
+          ({ symbol: tokenSymbol, chainId: tokenChainId }) =>
+            tokenSymbol === symbol && tokenChainId === chainId
+        ) ?? {}
+
+      if (
+        connector !== injected ||
+        !address ||
+        localStorage.getItem(storageKey) === 'true'
+      ) {
+        return
+      }
+
+      return ethereum
+        .request({
+          method: 'wallet_watchAsset',
+          params: {
+            type: 'ERC20',
+            options: {
+              address, // The address that the token is at.
+              symbol: tokenSymbol, // A ticker symbol or shorthand, up to 5 chars.
+              decimals, // The number of decimals in the token
+              image: logoURI // A string url of the token logo
+            }
+          }
+        })
+        .then(function () {
+          window.localStorage.setItem(storageKey, 'true')
+          // eslint-disable-next-line no-console
+          console.log('Token successfully registered on MetaMask.')
+        })
```

Comment:
Can's we just reuse `utils/wallet-watch-asset.js` and avoid duplicating code?

The two "claim" mini-apps, just call `watchAsset` after the user claims the tokens. Thinking that approach is simpler than using a hook.

Reply:
oh, I did not know it was there! In order to create a hook I used one as a reference in another project which was implemented as a component with null UI (meaning, it was returning null), so in that context a hook made more sense. Perhaps the only benefit of using the hook is that it makes it simpler to call it, as we can call directly `useWeb3React()` in the hook, instead of having to call it manually everytime we want to watch the asset.

current usage
```
// in component
const { account, chainId, connector } = useWeb3React()
watchAsset({ account, chainId, token: 'WETH' })
```

with the hook

```
const watchAsset = useRegisterToken({ symbol: 'WETH' })
watchAsset()
```

I find the latter a bit less verbose, but let me know your thoughts - I can drop it and keep the original function, or use the new function in the other places.

Reply:
As an intermediate solution, the hook may call `utils/wallet-watch-asset.js`. Thoughts?

Reply:
sounds ok for me! Here it is applied https://github.com/purefinance/pure.finance/pull/35/commits/5b76895ddaa75a6c746226a23e56871d52c3f5f5

Thanks for pointing out that utility function, I totally missed it!